src/cable.cr

```diff
@@ -32,8 +31,12 @@ module Cable
     setting token : String = "token", example: "token"
     setting url : String = ENV.fetch("REDIS_URL", "redis://localhost:6379"), example: "redis://localhost:6379"
     setting disable_sec_websocket_protocol_header : Bool = false
-    setting backend_class : Cable::BackendCore.class = Cable::RedisBackend, example: "Cable::RedisBackend"
-    setting redis_ping_interval : Time::Span = 15.seconds
+    setting backend_class : Cable::BackendCore.class = Cable::RegistryBackend, example: "Cable::RedisBackend"
+    setting backend_ping_interval : Time::Span = 15.seconds
+    @[Deprecated("Use backend_ping_interval")]
+    setting redis_ping_interval : Time::Span do
+      backend_ping_interval
+    end
```

Comment:
I can't tell if Habitat actually supports the block format like `getter` does, but it didn't raise an error.

Reply:
oh, that's interesting... At this level, `setting` is a macro that takes a TypeDeclaration. That basically does this..

```
def self.{{ type_dec.var }} : {{ type_dec.type }}
  {{ type_dec.value }}
end
```

A little more complex, but that basic idea. So I guess the question is, what does Crystal consider the block here? My guess is the block is passed to `setting` which never calls `yield` and so it's ignored :thinking:

Reply:
Yeah, there's an open issue on it. https://github.com/luckyframework/habitat/issues/54

Reply:
Very likely. When I looked at the code for the `setting` macro it looked like there was some indirection that I didn't know how to follow.

What I was aiming for here was something like the block for `getter` and `property` to lazily initialize the setting, so if you were using the Redis-specific setting name it would still work, but now that I write that out I'm thinking actually maybe we could just do methods for that instead of setting multiple ivars?

Reply:
Yeah, methods may be fine here. I'm updating habitat issues to track some of these things too.